engine/crates/engine-v2/schema/src/builder/ids.rs

```diff
@@ -30,11 +31,45 @@ where
     }
 }
 
-#[derive(Default)]
-pub struct IdMaps {
+pub(super) struct IdMaps {
```

Comment:
Initializing `IdMaps` (federated_graph ids -> engine_v2 ids) immediately with the right values with `new()`. Avoids the pitfall of trying to get the associated id but the idmap wasn't fully initialized yet.